Webdesign

samanthahamilton opened the pull request:
## PR for changes to web design and layout

### Files changed:  
- conf.py
- index.rst
- _new_ custom.css file

No associated Issue. 

### Message :

Hi @ianhi!

You mentioned before I could "mess around" with the Sphinx design. I made some changes and additions to the layout, let me know what you think. I used [Alabaster](https://alabaster.readthedocs.io/en/latest/) and [Write the Docs](https://www.writethedocs.org/) for thematic context. The colour scheme is based on Matpotlib's icon (HEX 11557C) and the blue/green of the python code. The background (HEX fafafa RBG 250) is a tone based off the Sphinx margin grey (HEX efefef RBG 239). 

Also, the `cmd` guide you left in the last PR worked, and my fork is now up to date with the repo 👍🏻 

Cheers,
Sam.

ianhi commented:
Hiya, sorry I wasn't able to look at this until now. I got rather carried away with the rest of life/work

> Also, the cmd guide you left in the last PR worked, and my fork is now up to date with the repo 👍🏻

:tada: :tada: rad :tada:  - I honestly wasn't entirely sure it would... so thanks for the followup

ianhi commented:
A few notes:
1. If you mouse over the items in TOC the text gets hidden:
![image](https://user-images.githubusercontent.com/10111092/94090567-eafbde80-fde3-11ea-9cd6-5d0780aeb2f2.png)
2. Thoughts on deviating from blue for links? I think what you switched too looks nicer but theres also like 30 years of expectation built around links being blue:
![image](https://user-images.githubusercontent.com/10111092/94090789-87be7c00-fde4-11ea-9bb7-cea374fa9e15.png)
3. This actually introduces two custom css files. One is already being injected here:
https://github.com/ianhi/mpl-interactions/blob/fe449ffacffa22c9702a098be7ac6e1daa6dd71b/docs/conf.py#L99-L100
(which I think is the more correct way? maybe... I think I got a warning when I tried to do it the way you did, and then I just copied something one of my friends had done)

So at the very least we'd need to fix the TOC mouse over. But overall I'm more ambivalent about these changes than I expected. I think that the RTD styling is nice because it is familiar to many users in the ecosystem. So maybe we shouldn't merge this unless there is a clear upside?

I will say that basing it off the matplotlib logo is super creative and I love the thought!

samanthahamilton commented:
Thanks for the feedback, I will take a look at these notes & changes on Monday!

samanthahamilton commented:
Hey @ianhi, I had some spare time to look over your comments today. Here are some thoughts:

> If you mouse over the items in TOC the text gets hidden

Fixed

>Thoughts on deviating from blue for links? I think what you switched too looks nicer but theres also like 30 years of expectation built around links being blue

I never thought of that, actually. Good point!! I changed them back to blue/purple.

> This actually introduces two custom css files.

I used this [procedure](https://docs.readthedocs.io/en/stable/guides/adding-custom-css.html#adding-custom-css-or-javascript-to-sphinx-documentation) to create the CSS file and place it in the `config` file. I didn't add that "extra" file... maybe it gets added when the build occurs?? But I don't know.

> overall I'm more ambivalent about these changes than I expected. I think that the RTD styling is nice because it is familiar to many users in the ecosystem. 

If you're a fan of the RTD styling then by all means stick with that! I have just finished a full stack web developer course, and this was a great "proof of principle" activity... and something that I can refer back to as practice. I just fell in love with the `Alabaster` theme, and was trying to do something similar. Feel free to keep the files for future reference -- use them or not, it's at your discretion. But playing around with them was great fun, and now I know I can make other changes you might need.

Best,
Sam.
